nncf/experimental/torch/nas/bootstrapNAS/training/progressive_shrinking_builder.py

```diff
@@ -48,7 +48,7 @@ def __init__(self, nncf_config: NNCFConfig, should_init: bool = True):
         super().__init__(nncf_config, should_init)
         bn_adapt_params = self._algo_config.get('batchnorm_adaptation', {})
         bn_adapt_algo_kwargs = get_bn_adapt_algo_kwargs(nncf_config, bn_adapt_params)
-        self._bn_adaptation = BatchnormAdaptationAlgorithm(**bn_adapt_algo_kwargs)
+        self._bn_adaptation = BatchnormAdaptationAlgorithm(**bn_adapt_algo_kwargs) if bn_adapt_algo_kwargs else None
```

Comment:
Again, this logic would break Training. There's lack of unit testing of case when `bn_adaptation` is disabled.

Currently, training will fail if bn_adaptation is empty, which would be not expected behavior.
```
    def _run_batchnorm_adaptation(self, model):
        if self._bn_adaptation is None:
            raise RuntimeError("Missing initialization of Batchnorm Adaptation algorithm")
        self._bn_adaptation.run(model)
```
Propose to verify behavior of Progressive Shrinking Controller in case of different `num_samples` for BNAdaptation.
Here's an example of unit test for Progressive Shrinking Controller : 
https://github.com/openvinotoolkit/nncf/blob/develop/tests/torch/nas/test_scheduler.py#L125
The test structure can be the same as it was suggested for SearchAlgorithm above.

Reply:
Thanks! Added test similar to search. 
In https://github.com/openvinotoolkit/nncf/blob/develop/nncf/experimental/torch/nas/bootstrapNAS/training/progressive_shrinking_builder.py#L50, we check Batchnorm adapt and number of samples. 
In ProgressiveShrinkingController we check if batchnorm adapt is enabled when preparing for validation and when setting stage, preventing to run _run_batchnorm_adaptation(..) if bn adapt is not enabled. 

We warn the user if _run_batchnorm_adaptation(..) is called directly, as well.